internal/nvidia_base/nvidia_base.go

```diff
@@ -0,0 +1,217 @@
+package nvidia_base
+
+// I'd prefer this to just be in nvidia/nvidia.go, but it can't, because that causes
+// a cyclic dependency, as sensors/nvidia.go and fans/nvidia.go need to call nvidia_base.GetDevice()
+// but nvidia/nvidia.go needs to import fans/nvidia.go and sensors/nvidia.go for nvidia.GetDevices().
+// So GetDevice() now is in this extra package that does not import any other internal code,
+// so everyone can import this.
+
+/* // some C code to use a function only available in some libnvidia-ml.so versions
```

Comment:
I just realized that this C code is actually used and executed, I thought this was just for reference since it is in the comments. I didn't know about this golang feature, interesting :smile:

Reply:
Yeah, it's amazing, but a bit wird: It must be in a comment that's directly before `import "C"`, you can't even have an empty line between the comment block and the import.
But it seems to work pretty well, nice solution if you only need relatively little C code for "glue"